tools/SourceKit/include/SourceKit/SwiftLang/Factory.h

```diff
@@ -0,0 +1,24 @@
+//===--- Factory.h - --------------------------------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_SOURCEKIT_SWIFTLANG_FACTORY_H
+#define LLVM_SOURCEKIT_SWIFTLANG_FACTORY_H
+
+namespace SourceKit {
+class LangSupport;
+class Context;
+
+std::unique_ptr<LangSupport> createSwiftLangSupport(Context &SKCtx);
```

Comment:
Headers should generally include the headers that they depend on, in this case it should get a `#include <memory>` (because of `std::unique_ptr`)